File: AspNetMvcUnleashed/0672329980_source/Chapter03/CS/MvcApplication1/MvcApplication1/Controllers/MerchandiseController.cs
using System.Web.Mvc;
using MvcApplication1.Models;

namespace MvcApplication1.Controllers
{
    public class MerchandiseController : Controller
    {
        private MerchandiseRepository _repository = new MerchandiseRepository();

        // GET: /Merchandise/Edit
        [ActionName("Edit")]
        [AcceptVerbs(HttpVerbs.Get)]
        public ActionResult Edit_GET(Merchandise merchandiseToEdit)
        {
            return View(merchandiseToEdit);
        }

        // POST: /Merchandise/Edit
        [ActionName("Edit")]
        [AcceptVerbs(HttpVerbs.Post)]
        public ActionResult Edit_POST(Merchandise merchandiseToEdit)
        {
            try
            {
                _repository.Edit(merchandiseToEdit);
                return RedirectToAction("Edit");
            }
            catch
            {
                return View();
            }
        }
    }
}

File: AspNetMvcUnleashed/0672329980_source/Chapter03/VB/MvcApplication1/MvcApplication1/Controllers/MerchandiseController.vb
Public Class MerchandiseController
    Inherits System.Web.Mvc.Controller

    Private _repository As New MerchandiseRepository()

    ' GET: /Merchandise/Edit
    <ActionName("Edit")> _
    <AcceptVerbs(HttpVerbs.Get)> _
    Function Edit_GET(ByVal merchandiseToEdit As Merchandise) As ActionResult
        Return View(merchandiseToEdit)
    End Function

    ' POST: /Merchandise/Edit
    <ActionName("Edit")> _
    <AcceptVerbs(HttpVerbs.Post)> _
    Function Edit_POST(ByVal merchandiseToEdit As Merchandise) As ActionResult
        Try
            _repository.Edit(merchandiseToEdit)
            Return RedirectToAction("Edit")
        Catch
            Return View()
        End Try
    End Function

End Class
